crud/compare/filters.lua

```diff
@@ -17,12 +17,8 @@ local filters = {}
     - opposite condition `'id' < 100` becomes false
     - in such case we can exit from iteration
 ]]
-local function is_early_exit_possible(index, tarantool_iter, condition)
-    if index == nil then
-        return false
-    end
-
-    if index.name ~= condition.operand then
+local function is_early_exit_possible(index, scan_index, tarantool_iter, condition)
+    if index ~= scan_index then
```

Comment:
To avoid problems in the future (an index may be `deepcopy'ed` in some place).
```suggestion
    if index.name ~= scan_index.name then
```

Reply:
Reworked